cocos/2d/CCNode.h

```diff
@@ -1940,8 +1940,23 @@ class CC_DLL Node : public Ref
     mutable bool _additionalTransformDirty; ///< transform dirty ?
     bool _transformUpdated;         ///< Whether or not the Transform object was updated since the last frame
 
-    std::int64_t _localZOrderAndArrival; /// cache, for 64bits compress optimize.
-    int _localZOrder; /// < Local order (relative to its siblings) used to sort the node
+#if CC_LITTLE_ENDIAN
```

Comment:
I am not confident with `CC_LITTLE_ENDIAN ` computation. And i am not confident with union usage here. Does `int` and `unsigned int` should be 32 bit on all platforms? I think it is better to use two variables here.

Reply:
Actual use union structure, the implement source code is more clearly and easy, and the access code is not changed by original code, As far as I know, all platforms: int is 32 bit. Or can use std::int32_t to replace it.

Reply:
Yep, use `std::int32_t` is more safe.

Reply:
Anonymous union appears to be GNU extension, though Clang appears to support it just fine. Not sure about MSVC.